servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpLoadBalancerFactory.java

```diff
@@ -199,19 +371,44 @@ public Completable closeAsync() {
             return delegate.closeAsync();
         }
 
-        @Override
-        public Completable closeAsyncGracefully() {
-            return delegate.closeAsyncGracefully();
-        }
-
         @Override
         public StreamingHttpRequest newRequest(final HttpRequestMethod method, final String requestTarget) {
             return delegate.newRequest(method, requestTarget);
         }
 
-        @Override
-        public String toString() {
-            return delegate.toString();
+        private static final class AtMostOnceDeliveryRequestTracker implements RequestTracker {
+
+            private static final AtomicIntegerFieldUpdater<AtMostOnceDeliveryRequestTracker> doneUpdater =
+                    newUpdater(AtMostOnceDeliveryRequestTracker.class, "done");
+
+            private final RequestTracker original;
+            private volatile int done;
+
+            private AtMostOnceDeliveryRequestTracker(final RequestTracker original) {
+                this.original = original;
+            }
+
+            @Override
+            public long beforeStart() {
+                if (doneUpdater.compareAndSet(this, 0, 1)) {
```

Comment:
Is this right? It seems like we can only call one of the three methods but we really need to call `beforeStart()` once then one of the other two exactly once after that.

Reply:
yeah it was late :-) 
I fixed it but never pushed.